diesel/src/mysql/query_builder/query_fragment_impls.rs

```diff
@@ -63,3 +69,71 @@ where
         Ok(())
     }
 }
+
+impl<T> QueryFragment<Mysql, crate::mysql::backend::MysqlOnConflictClause> for DoNothing<T>
+where
+    T: Table,
+    T::PrimaryKey: Column,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, Mysql>) -> QueryResult<()> {
+        out.push_sql(" UPDATE ");
+        out.push_identifier(<T::PrimaryKey as Column>::NAME)?;
+        out.push_sql(" = ");
+        out.push_identifier(<T::PrimaryKey as Column>::NAME)?;
+        Ok(())
+    }
+}
+
+impl<T, Tab> QueryFragment<Mysql, crate::mysql::backend::MysqlOnConflictClause> for DoUpdate<T, Tab>
+where
+    T: QueryFragment<Mysql>,
+    Tab: Table,
+    Tab::PrimaryKey: Column,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, Mysql>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+        out.push_sql(" UPDATE ");
+        if self.changeset.is_noop(out.backend())? {
+            out.push_identifier(<Tab::PrimaryKey as Column>::NAME)?;
+            out.push_sql(" = ");
+            out.push_identifier(<Tab::PrimaryKey as Column>::NAME)?;
+        } else {
+            self.changeset.walk_ast(out.reborrow())?;
+        }
+        Ok(())
+    }
+}
+
+impl<Values, Target, Action> QueryFragment<Mysql, MysqlOnConflictClause>
+    for OnConflictValues<Values, Target, Action, NoWhereClause>
+where
+    Values: QueryFragment<Mysql>,
+    Target: QueryFragment<Mysql>,
+    Action: QueryFragment<Mysql>,
+    NoWhereClause: QueryFragment<Mysql>,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, Mysql>) -> QueryResult<()> {
+        self.values.walk_ast(out.reborrow())?;
+        out.push_sql(" ON DUPLICATE KEY");
+        self.target.walk_ast(out.reborrow())?;
+        self.action.walk_ast(out.reborrow())?;
+        self.where_clause.walk_ast(out)?;
```

Comment:
Because this implementation is marked with `NoWhereClause` means that this like effectively does nothing?

Reply:
Yes, although I prefer to call that function nevertheless for the so that if we ever add something in that `QueryFragment` impl we get it here as well.